ufl/algorithms/compute_form_data.py

```diff
@@ -386,6 +404,13 @@ def compute_form_data(form,
     # faster!
     preprocessed_form = reconstruct_form_from_integral_data(self.integral_data)
     check_form_arity(preprocessed_form, self.original_form.arguments())  # Currently testing how fast this is
+    
+    # Optimises complex nodes by removing e.g. Conj(Conj(...)) if in complex mode
+    # Otherwise removes complex nodes entirely.
+    if complex_mode:
+        preprocessed_form = optimise_complex_nodes(preprocessed_form)
```

Comment:
But if so, then we would want to apply `do_comparison_check` again here.

Reply:
The optimise pass doesn't remove single real nodes so this isn't required.

Reply:
Perhaps `optimise_complex_nodes` should be `apply_complex_simplifications`?

Reply:
> Perhaps `optimise_complex_nodes` should be `apply_complex_simplifications`?

:+1: 

We should also consider doing these simplifications on the spot, that is, in the `__new__` method of the appropriate operators.